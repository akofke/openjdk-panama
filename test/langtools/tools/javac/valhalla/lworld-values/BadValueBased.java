/*
 * @test /nodynamiccopyright/
 * @summary Check that improper application of "@ValueBased" annotation are caught
 * @compile/fail/ref=BadValueBased.out -XDrawDiagnostics -XDdev BadValueBased.java
 */

public class BadValueBased {
    @ValueBased 
    interface X {}

    @ValueBased 
    @interface A {}

    @ValueBased
    enum E {}

    @ValueBased 
    class Y {
        @ValueBased int x;
    }
}
