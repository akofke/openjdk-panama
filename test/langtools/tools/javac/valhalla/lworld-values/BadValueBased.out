BadValueBased.java:8:5: compiler.err.bad.value.based.anno
BadValueBased.java:11:5: compiler.err.bad.value.based.anno
BadValueBased.java:14:5: compiler.err.bad.value.based.anno
BadValueBased.java:19:9: compiler.err.annotation.type.not.applicable
4 errors
